### PR TITLE
Adding auto-dotenv recipe

### PR DESCRIPTION
**Pull request by apoorvkh**
This is a super lightweight recipe that automatically loads environment variables from `.env` after `conda activate` is called. The previous variables are restored when the environment is deactivated.

### Motivation

Some projects need users to set certain environment variables (commonly stored in a `.env` file). This is a convenience package that guarantees the environment variables will be loaded when the environment is activated. This also saves the user one command (i.e. running `source .env` themselves) every session.

Normally, conda [supports](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#setting-environment-variables) including `variables:` entries in `environment.yml`. But micromamba doesn't actually implement this (https://github.com/mamba-org/mamba/issues/1026). And perhaps `.env` is more conventional. This package provides a workaround.

### Other details

This is currently implemented for Linux and supports bash/zsh shells. We can consider adding scripts for csh/fish later. The script is currently pretty straightforward and I have tested it manually (e.g. by creating a `.env` file, loading and unloading environments, etc). There is no source or URL for these scripts.

Hope that sounds good. Thanks!

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by apoorvkh** [comment]
@conda-forge-admin, please ping conda-forge/staged-recipes

Ready for review, thanks!

File: recipes/auto-dotenv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package:
+  name: auto-dotenv
+  version: 1.0
+
+build:
+  skip: true  # [win]
```
Comment (ocefpaf) [inline recipes/auto-dotenv/meta.yaml:6]: My guess is that this one could be noarch generic and installable on Windows for those who use the Windows bash, right?

Comment (apoorvkh) [reply recipes/auto-dotenv/meta.yaml]: That seems right to me! I updated to `noarch: generic`.

**Comment by apoorvkh** [comment]
Thank you!

**Pull request merged**